compiler/rustc_hir_typeck/src/method/probe.rs

```diff
@@ -1415,6 +1418,32 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
                         self.param_env,
                         impl_bounds,
                     ));
+
+                    for obligation in obligations {
+                        if self.infcx.next_trait_solver() {
+                            ocx.register_obligation(obligation);
+                        } else {
+                            match self.infcx.evaluate_obligation_no_overflow(&obligation) {
+                                EvaluationResult::EvaluatedToOk => {
+                                    // No side-effects, no need to register obligations.
+                                }
+                                EvaluationResult::EvaluatedToOkModuloRegions
```

Comment:
TODO: We could also skip registering the obligations if they're OkModuloRegions, since we already check for leak check failures in the eval call.